### PR TITLE
[1.17] Add default env configuration

### PR DESCRIPTION
**Pull request by mrunalp**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR adds configuration option to inject default environment variables into all containers
started by cri-o. If the container image or runtime options specify the same env variable, then it gets overridden. 

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
Add configuration for setting default environment in all containers.
```

### CODE REVIEW

File: internal/pkg/criocli/criocli.go [hunk 1]
```diff
@@ -208,6 +208,9 @@ func mergeConfig(config *libconfig.Config, ctx *cli.Context) (string, error) {
 	if ctx.IsSet("conmon-env") {
 		config.ConmonEnv = ctx.StringSlice("conmon-env")
 	}
+	if ctx.IsSet("default-env") {
+		config.ConmonEnv = ctx.StringSlice("default-env")
```
Comment (rphillips) [inline internal/pkg/criocli/criocli.go:212]: s/ConmonEnv/DefaultEnv

Comment (mrunalp) [reply internal/pkg/criocli/criocli.go:212]: yeah, fixing!

File: completions/fish/crio.fish [hunk 2]
```diff
@@ -95,13 +96,13 @@ complete -c crio -n '__fish_crio_no_subcommand' -f -l profile -d 'Enable pprof r
 complete -c crio -n '__fish_crio_no_subcommand' -f -l profile-port -r -d 'Port for the pprof profiler'
 complete -c crio -n '__fish_crio_no_subcommand' -f -l read-only -d 'Setup all unprivileged containers to run as read-only. Automatically mounts tmpfs on `/run`, `/tmp` and `/var/tmp`. (default: false)'
 complete -c crio -n '__fish_crio_no_subcommand' -f -l registry -r -d 'Registry to be prepended when pulling unqualified images, can be specified multiple times'
-complete -c crio -n '__fish_crio_no_subcommand' -l root -s r -r -d 'The CRI-O root directory (default: "/var/lib/containers/storage")'
-complete -c crio -n '__fish_crio_no_subcommand' -l runroot -r -d 'The CRI-O state directory (default: "/var/run/containers/storage")'
+complete -c crio -n '__fish_crio_no_subcommand' -l root -s r -r -d 'The CRI-O root directory (default: "/home/mrunalp/.local/share/containers/storage")'
```
Comment (TomSweeneyRedHat) [inline completions/fish/crio.fish:99]: I'm thinking the mrunalp reference is probably incorrect.

Comment (mrunalp) [reply completions/fish/crio.fish]: Fixed.

**Comment by mrunalp** [comment]
@haircommander @umohnani8 ptal

**Pull request merged**